### PR TITLE
Adds TQDMProgressBar in callbacks

### PR DESCRIPTION
**Pull request by shun-lin**
Adding TQDMProgressBar in callbacks to make integration of TQDM (for progress bar) with tensorflow easy and configurable.

I also included ipython notebook usage / demo in the docs/tutorial folder.

Original source code:
https://github.com/bstriner/keras-tqdm
https://gist.github.com/ebursztein/5ff21d7600556de9946eac4a313344be

Original Pull Request :
https://github.com/tensorflow/addons/pull/519
(sorry I messed up with merging and thus I have create another repo, the current one, to make pull request).

Based on the comments on 519:
1. I have ran ```bash tools/run_docker.sh -c 'make code-format'``` and all tests (3) passes.
2. I have updated callback README.

Thank you so much for your time!

### CODE REVIEW

**Comment by shun-lin** [comment]
@seanpmorgan I have fixed a typo that is causing tests to fail, may you add the ```kokoro:force-run``` label again? Thank you so much for your time!

**Comment by seanpmorgan** [comment]
> @seanpmorgan I have fixed a typo that is causing tests to fail, may you add the `kokoro:force-run` label again? Thank you so much for your time!

Hmmmm looks like there is still a formatting error that is found. Can you double check that any re-formats were committed/pushed

**Comment by shun-lin** [comment]
> > @seanpmorgan I have fixed a typo that is causing tests to fail, may you add the `kokoro:force-run` label again? Thank you so much for your time!
> 
> Hmmmm looks like there is still a formatting error that is found. Can you double check that any re-formats were committed/pushed

@seanpmorgan sorry I forgot to check sanity with ```bash tools/run_docker.sh -c 'make sanity-check'``` I have ran it and updated the code to pass all the sanity check, may you add `kokoro:force-run` label again? Thank you so much for your time!

Another question, I found out that when someone just look at the notebook, the private output of the kernal does not show the progress bar, where should I put the below screenshot (or if I should at all)? If it would be helpful, I am thinking of putting it under `doc/tutorial/assets` and then link it in the notebook. Thank you so much! 
https://drive.google.com/file/d/1EEqoF37tknt5Gtl5GCp9_wDh5CRDqF15/view?usp=sharing

**Comment by shun-lin** [comment]
@Squadrick when you have time may you review this PR? Thank you so much for your time!

File: tensorflow_addons/callbacks/README.md [hunk 1]
```diff
@@ -3,12 +3,12 @@
 ## Maintainers
 | Submodule  | Maintainers  | Contact Info   |
 |:---------- |:------------- |:--------------|
-|  | | |
+|  tqdm_progress_bar | @shun-lin | shunlin@google.com |
 
 ## Contents
 | Submodule | Metric  | Reference                               |
```
Comment (seanpmorgan) [inline tensorflow_addons/callbacks/README.md]: Could you edit this README to say `Callback` instead of `Metric` then you can fill it in with `TQDMProgressBar`

File: docs/tutorials/tqdm_progress_bar.ipynb [hunk 2]
```diff
@@ -0,0 +1,465 @@
+{
```
Comment (seanpmorgan) [inline docs/tutorials/tqdm_progress_bar.ipynb]: <p>Mind removing these warnings from the notebook</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/610/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (seanpmorgan) [inline docs/tutorials/tqdm_progress_bar.ipynb]: <p>Hmmm is this what the expected output looks like? I thought it should show similar to:</p><p><a href="https://i.stack.imgur.com/gpSTt.png" target="_blank">https://i.stack.imgur.com/gpSTt.png</a></p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/610/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

**Comment by seanpmorgan** [comment]
Also just want to tag @bstriner. The repo has an MIT license so I think it's fine but wanted to tag for visibility

File: tensorflow_addons/callbacks/README.md (hunk 1 above)
Comment (shun-lin) [reply tensorflow_addons/callbacks/README.md]: got it :)

File: docs/tutorials/tqdm_progress_bar.ipynb (hunk 2 above)
Comment (shun-lin) [reply docs/tutorials/tqdm_progress_bar.ipynb]: Will do, thanks!

**Comment by shun-lin** [comment]
> Also just want to tag @bstriner. The repo has an MIT license so I think it's fine but wanted to tag for visibility

Thanks Sean!

File: docs/tutorials/tqdm_progress_bar.ipynb (hunk 2 above)
Comment (shun-lin) [reply docs/tutorials/tqdm_progress_bar.ipynb]: Resolved, see the newest commit for the change!

**Comment by shun-lin** [comment]
@seanpmorgan I have updated this Pull Request based on your comments, may you run ```kokoro:force-run``` again and review the PR? Thank you so much for your time!

File: docs/tutorials/tqdm_progress_bar.ipynb [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
```
Comment (seanpmorgan) [inline docs/tutorials/tqdm_progress_bar.ipynb]: <p>Could you add <code>!pip install -q --no-deps tensorflow-addons~=0.6</code> we'll use this notebook as the test for the callback so the notebook will need to be able to execute</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/610/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (seanpmorgan) [inline docs/tutorials/tqdm_progress_bar.ipynb]: <p>Instead of merging this .png could you just display the URL using markdown?</p><p> </p><p><code>&nbsp;&nbsp;"fig"</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/610/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

File: tensorflow_addons/callbacks/tqdm_progress_bar.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+"""TQDM Progress Bar. """
+
+from __future__ import absolute_import, division, print_function
+
+import time
+from collections import defaultdict
+
+from tensorflow.keras.callbacks import Callback
+from tensorflow_addons.utils import keras_utils
+from tqdm.auto import tqdm
```
Comment (seanpmorgan) [inline tensorflow_addons/callbacks/tqdm_progress_bar.py]: Since tqdm isn't a requirement for addons could you put this import as a try-catch within the class that returns an error if tqdm is unable to import

File: docs/tutorials/tqdm_progress_bar.ipynb (hunk 3 above)
Comment (seanpmorgan) [inline docs/tutorials/tqdm_progress_bar.ipynb]: <p>Can you set this to use TF2:</p><p><code></code><code></p><p>try:</p><p>&nbsp;%tensorflow_version 2.x</p><p>except:</p><p>&nbsp;pass</p><p> </p><p>import tensorflow as tf</p><p></code><code></code></p><p> </p><p>We should be able to remove these in the near future</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/610/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

File: tensorflow_addons/callbacks/tqdm_progress_bar.py (hunk 4 above)
Comment (shun-lin) [reply tensorflow_addons/callbacks/tqdm_progress_bar.py]: got it, will do!

**Comment by shun-lin** [comment]
> Thanks for the updates. So I tried running this in colab and didn't return the expected result. Any ideas?
> https://colab.research.google.com/drive/1yIRkyVZEbOWtAzr2BKDoyDeCJFJinR7d

I will look into it asap, thanks!

File: docs/tutorials/tqdm_progress_bar.ipynb (hunk 3 above)
Comment (shun-lin) [reply docs/tutorials/tqdm_progress_bar.ipynb]: @seanpmorgan I thought this snippet fetched tensorflow 2.0

Copied from the tensorflow 2.0 for expert codelab
https://colab.sandbox.google.com/github/tensorflow/docs/blob/master/site/en/tutorials/quickstart/advanced.ipynb

Comment (shun-lin) [reply docs/tutorials/tqdm_progress_bar.ipynb]: got it!

Comment (shun-lin) [reply docs/tutorials/tqdm_progress_bar.ipynb]: got it!

Comment (shun-lin) [reply docs/tutorials/tqdm_progress_bar.ipynb]: updated in the latest patch!

**Comment by shun-lin** [comment]
@seanpmorgan I have updated this PR based on your comments and found the solution to the issue mentioned (```tqdm>=4.36.1``` solves it). When you have time may you review it again? Thank you so much for your time!

File: docs/tutorials/tqdm_progress_bar.ipynb (hunk 3 above)
Comment (shun-lin) [reply docs/tutorials/tqdm_progress_bar.ipynb]: updated in latest patch!

File: tensorflow_addons/callbacks/tqdm_progress_bar.py (hunk 4 above)
Comment (shun-lin) [reply tensorflow_addons/callbacks/tqdm_progress_bar.py]: updated in latest patch!

File: tensorflow_addons/callbacks/tqdm_progress_bar.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+from __future__ import absolute_import, division, print_function
+
+import time
+from collections import defaultdict
+
+from tensorflow.keras.callbacks import Callback
+from tensorflow_addons.utils import keras_utils
+
+try:
+    from tqdm.auto import tqdm
```
[the remark this answers is not in the document]
Comment (seanpmorgan) [reply tensorflow_addons/callbacks/tqdm_progress_bar.py]: Also could you move this into the `__init__` of the class. Because all callbacks are imported at import this will cause issues for users that do not intend to user this:
